modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNestjsClientCodegen.java

```diff
@@ -0,0 +1,603 @@
+            }
+        }
+
+        if (additionalProperties.containsKey(WITH_INTERFACES)) {
+            boolean withInterfaces = Boolean.parseBoolean(additionalProperties.get(WITH_INTERFACES).toString());
+            if (withInterfaces) {
+                apiTemplateFiles.put("apiInterface.mustache", "Interface.ts");
+            }
+        }
+
+        if (additionalProperties.containsKey(TAGGED_UNIONS)) {
+            taggedUnions = Boolean.parseBoolean(additionalProperties.get(TAGGED_UNIONS).toString());
+        }
+
+        if (nestVersion.atLeast("6.0.0")) {
+            if (!additionalProperties.containsKey(PROVIDED_IN_ROOT)) {
+                additionalProperties.put(PROVIDED_IN_ROOT, true);
+            } else {
+                additionalProperties.put(PROVIDED_IN_ROOT, Boolean.parseBoolean(
+                    additionalProperties.get(PROVIDED_IN_ROOT).toString()
+                ));
+            }
+        } else {
+            additionalProperties.put(PROVIDED_IN_ROOT, false);
+        }
+
+        additionalProperties.put(NEST_VERSION, nestVersion);
+        additionalProperties.put("injectionToken", nestVersion.atLeast("4.0.0") ? "InjectionToken" : "OpaqueToken");
+        additionalProperties.put("injectionTokenTyped", nestVersion.atLeast("4.0.0"));
+        additionalProperties.put("useHttpClient", nestVersion.atLeast("4.3.0"));
+        additionalProperties.put("useRxJS6", nestVersion.atLeast("6.0.0"));        
+        if (additionalProperties.containsKey(SERVICE_SUFFIX)) {
+            serviceSuffix = additionalProperties.get(SERVICE_SUFFIX).toString();
+            validateClassSuffixArgument("Service", serviceSuffix);
+        }
+        if (additionalProperties.containsKey(SERVICE_FILE_SUFFIX)) {
+            serviceFileSuffix = additionalProperties.get(SERVICE_FILE_SUFFIX).toString();
+            validateFileSuffixArgument("Service", serviceFileSuffix);
+        }
+        if (additionalProperties.containsKey(MODEL_SUFFIX)) {
+            modelSuffix = additionalProperties.get(MODEL_SUFFIX).toString();
+            validateClassSuffixArgument("Model", modelSuffix);
+        }
+        if (additionalProperties.containsKey(MODEL_FILE_SUFFIX)) {
+            modelFileSuffix = additionalProperties.get(MODEL_FILE_SUFFIX).toString();
+            validateFileSuffixArgument("Model", modelFileSuffix);
+        }
+        if (additionalProperties.containsKey(FILE_NAMING)) {
+            this.setFileNaming(additionalProperties.get(FILE_NAMING).toString());
+        }
+    }
+
+    private void addNpmPackageGeneration(SemVer nestVersion) {
+
+        if (additionalProperties.containsKey(NPM_REPOSITORY)) {
+            this.setNpmRepository(additionalProperties.get(NPM_REPOSITORY).toString());
+        }
+
+        // Set the typescript version compatible to the Nestjs version
+        if (nestVersion.atLeast("6.0.0")) {
+            additionalProperties.put("tsVersion", ">=3.4.0 <3.6.0");
```

Comment:
why `<3.6.0`? We're successfully running Nest 6 with TS 3.7.5 and I'm not aware of any requirements that would limit the version to `<3.6`.

Reply:
removed set to ">=3.4.0 <4.0.0"

Reply:
well, does it break with 4.0? :D (if not, why use max version limit?)